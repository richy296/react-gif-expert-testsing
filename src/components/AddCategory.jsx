import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const newInputValue = inputValue.trim();
        if (newInputValue.length <= 1) return;
        
        //setCategories(categories => [inputValue, ...categories]);
        onNewCategory(newInputValue)
        setInputValue('');
    }

    return (
        <form aria-label="form" onSubmit={onSubmit}>
            <input 
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}