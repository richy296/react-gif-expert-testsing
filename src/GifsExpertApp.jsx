import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GifsExpertApp = () => {

    const [categories, setCategories] = useState(['Dbz']);

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;
        setCategories(cat => [newCategory, ...cat])
    }
 
    return (
        <>
            {/* Titulo */}
            <h1>GifsExpertApp</h1>

            {/* Input */}
            <AddCategory 
                onNewCategory = {(newCategory) => onAddCategory(newCategory)}
            />

            {/* Listado de Gif */}
            {/* <button onClick={onAddCategory}>Agregar</button> */}
            { 
                categories.map((category) => (
                    <GifGrid key={category} category={category} />
                ))
            }
        </>
    )
}
