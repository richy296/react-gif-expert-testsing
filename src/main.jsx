import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifsExpertApp } from './GifsExpertApp';
import  './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifsExpertApp />
  </React.StrictMode>,
)
