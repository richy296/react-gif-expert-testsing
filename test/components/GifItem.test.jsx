import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Test in <GifItem />', () => { 
    const title = 'Goku';
    const url = 'https://gif.icon.com/goku.jpg';
    test('should make match with the snapshot', () => {
        const  {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    })    

    test('should show the title "Goku"', () => {
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('should show the image with the url and ult created', () => {
        render(<GifItem title={title} url={url}/>)
        //screen.debug();
        //expect(screen.getByRole('img').src).toBe(url);
        //expect(screen.getByRole('img').alt).toBe(title);

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

})