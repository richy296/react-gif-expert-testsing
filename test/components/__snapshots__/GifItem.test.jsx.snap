// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem /> should make match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      aria-label="Goku"
      src="https://gif.icon.com/goku.jpg"
      title="Goku"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
